lib/TypeNames/ModelTypeNames.cpp

```diff
@@ -229,38 +229,9 @@ TypeString PCTB::getNamedCInstance(const model::Type &Type,
   return TypeString(std::move(Result));
 }
 
-static RecursiveCoroutine<std::string>
-getArrayWrapperImpl(const model::Type &Type, const ptml::CBuilder &B) {
-  if (auto *Array = llvm::dyn_cast<model::ArrayType>(&Type)) {
-    std::string Result = "array_" + std::to_string(Array->ElementCount())
-                         + "_of_";
-    Result += rc_recur getArrayWrapperImpl(*Array->ElementType(), B);
-    rc_return Result;
-
-  } else if (auto *D = llvm::dyn_cast<model::DefinedType>(&Type)) {
-    std::string Result = (D->IsConst() ? "const_" : "");
-    rc_return std::move(Result += D->unwrap().name().str().str());
-
-  } else if (auto *Pointer = llvm::dyn_cast<model::PointerType>(&Type)) {
-    std::string Result = (D->IsConst() ? "const_ptr_to_" : "ptr_to_");
-    rc_return std::move(Result += rc_recur
-                          getArrayWrapperImpl(*Pointer->PointeeType(), B));
-
-  } else if (auto *Primitive = llvm::dyn_cast<model::PrimitiveType>(&Type)) {
-    std::string Result = (D->IsConst() ? "const_" : "");
-    rc_return std::move(Result += Primitive->getCName());
-
-  } else {
-    revng_abort("Unsupported model::Type.");
-  }
-}
-
 TypeString PCTB::getArrayWrapper(const model::ArrayType &ArrayType) const {
-  std::string Result(artificialArrayWrapperPrefix());
-
-  Result += getArrayWrapperImpl(ArrayType, *this);
-
-  return TypeString(getTag(ptml::tags::Span, std::move(Result)).toString());
+  auto Name = NamingHelper.artificialArrayWrapper(ArrayType);
+  return TypeString(getTag(ptml::tags::Span, std::move(Name)).toString());
```

Comment:
Shouldn't the call to `getTag` live in the body of `artificialArrayWrapper`?

Inthe current form, someone reading this needs to look at the body of `artificialArrayWrapper` anyway to understand what's going on.

Reply:
mmh no you are right we don't want to couple the naming with PTML.

yeah add `Name` please. and forget my complains about `getTag`.

Reply:
Done!